qmk_cli/subcommands/console.py

```diff
@@ -141,7 +141,7 @@ def run_forever(self):
                 identifier = (int2hex(message['vendor_id']), int2hex(message['product_id'])) if self.numeric else (message['manufacturer_string'], message['product_string'])
                 message['identifier'] = ':'.join(identifier)
                 message['ts'] = '{style_dim}{fg_green}%s{style_reset_all} ' % (strftime(cli.config.general.datetime_fmt),) if cli.args.timestamp else ''
-
+                message['text'] = message['text'].replace('%', '%%')
                 cli.echo('%(ts)s%(color)s%(identifier)s:%(index)d{style_reset_all}: %(text)s' % message)
```

Comment:
Another attempt — because named format specifiers don't work due to name collisions, use positional ones instead:
```suggestion

                cli.echo('%s', '%(ts)s%(color)s%(identifier)s:%(index)d{style_reset_all}: %(text)s' % message)
```
Just escaping `message['text']` might not be enough, because `message['identifier']` may also contain some `%` characters.  Actually there is the `spaceman/2_milk` keyboard in the repo, which has `"keyboard_name": "2% Milk"`, and several variants of `Clueboard 60%` and `Clueboard 66%`.  Those keyboard names can cause a crash in some Xorg versions though: https://gitlab.freedesktop.org/xorg/xserver/-/issues/845.

Reply:
(The mentioned Xorg crash should be fixed now: https://gitlab.freedesktop.org/libinput/libinput/-/issues/752)

Reply:
This one works.